oidc-authentication.md

```diff
@@ -241,10 +241,119 @@ As defined by [OIDC]:
+
+The Solid Project (and similar decentralized personal data servers) use
+[authentication]() for the following purposes:
+
+1. **To enable authorization**. First and foremost, authentication is used as a
+  mechanism to enable rule-based access control (WAC) (although see Section
+  2.1.1). Authorization requires knowing _both_ the identity of the user and the
+  identity of the client, and as a result, authentication needs to provide
+  (whenever technically possible) a tuple of: *(controller id, client id)*. To
+  put it another way, for any given action (such as an [authenticated
+  request]() to a data store), the access control mechanism needs to know _who_
+  performed the action, and _what software/client_ they were using to perform
+  it.
+
+2. **To enable discovery**. Resolving a controller's or a client's identifier
+  results in an [identifier document]() which is a useful starting point the
+  discovery of user and client metadata. For example, a client application
+  frequently requires some UI elements with which to address the user (a
+  username or a profile picture), as well as the user's preferences and other
+  data relevant to this app.
+
+3. **For auditing**. Many use cases require the keeping of audit trails and
+  access logs, which requires knowing who and what accessed a given resource or
+  performed an action.
+
+### 2.1.1 Authorization Without Authentication
+
+Authorization systems sometimes require the ability to perform access control
+decisions _without_ authentication, without identifying who is performing an
+action, either because the protected resource is sufficiently low-value (or
+low-threat if stolen), or due to practical requirements such as when sharing an
+online document for dissemination and collaboration with an unknown audience.
+
+As a result, most deployed authorization systems tend to employ a _hybrid_
+paradigm: they allow both identity-based and capability-based access control.
+Systems that prefer strict identity based Access Control Lists nevertheless
+develop capability links (unguessable tokens or URLs that allow access to the
+resource). Similarly, even systems such as [Authorization Capabilities]()
+(originally known as Object Capabilities) that are staunchly _against_
+authentication or using identifiers for access control, are forced to
+accommodate authentication requirements (if only for purposes of auditing) --
+see, for example, the [Horton protocol](http://www.erights.org/download/horton/document.pdf)
+used by the object capabilities community.
+
+Although authorization is largely out of scope for this document, it is worth
+mentioning that this specification aims to support this hybrid approach, and
+provide functionality for authentication as well as token-based capabilities.
 
 ## 2.2 Identity
 
+For a given [entity]() (whether a person, an organization or a software agent),
+the term [identity]() refers to a combination of: an [identifier]() (such as a
```

Comment:
What would entity and identity link to?

Reply:
When we convert to Bikeshed, all these empty links will auto-link to the definitions in the Terminology section.